infinispan-remote/src/test/java/org/hibernate/ogm/datastore/infinispanremote/utils/TransactionalRemoteCacheCounter.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.infinispanremote.utils;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.hibernate.Session;
+import org.hibernate.engine.spi.CollectionEntry;
+import org.hibernate.engine.spi.EntityKey;
+import org.hibernate.engine.spi.PersistenceContext;
+import org.hibernate.metamodel.spi.MetamodelImplementor;
+import org.hibernate.ogm.datastore.infinispanremote.impl.ProtoStreamMappingAdapter;
+import org.hibernate.ogm.datastore.infinispanremote.impl.protostream.ProtostreamId;
+import org.hibernate.ogm.datastore.infinispanremote.impl.protostream.ProtostreamPayload;
+import org.hibernate.ogm.hibernatecore.impl.OgmSessionImpl;
+import org.hibernate.ogm.model.impl.EntityKeyBuilder;
+import org.hibernate.ogm.model.key.spi.RowKey;
+import org.hibernate.ogm.persister.impl.OgmCollectionPersister;
+import org.hibernate.ogm.persister.impl.OgmEntityPersister;
+import org.hibernate.persister.entity.EntityPersister;
+
+import org.infinispan.client.hotrod.impl.transaction.TransactionalRemoteCacheImpl;
+
+/**
+ * Counts the entities and associations where {@link org.infinispan.client.hotrod.RemoteCache} is Transactional.
+ * <p>
+ * We can't use {@link TransactionalRemoteCacheImpl#size()} or query APIs,
+ * because they aren't transactional, because they don't interact with the transaction's data.
+ * <p>
+ * The algorithm is in not suitable for production code.
+ * Used in {@link InfinispanRemoteTestHelper} with a test scope.
+ *
+ * @author Fabio Massimo Ercoli
+ * @see TransactionalRemoteCacheImpl
+ */
+public class TransactionalRemoteCacheCounter {
+
+	public static int count(OgmEntityPersister persister, TransactionalRemoteCacheImpl cache, ProtoStreamMappingAdapter mapper,
+			Session currentSession) {
+		if ( cache.getTransactionManager() == null ) {
+			return cache.size();
+		}
+
+		final Set<Object> committedKeys = new HashSet<>();
+		cache.keySet().forEach( committedKey -> {
+			Boolean stillPresentInLocalTransaction = mapper.withinCacheEncodingContext( c -> c.containsKey( committedKey ) );
+			if ( stillPresentInLocalTransaction ) {
+				// skip count of removed entries
+				committedKeys.add( committedKey );
+			}
+		} );
+
+		// count all committed keys not removed by the local transaction
+		Integer counter = new Integer( committedKeys.size() );
```

Comment:
Do you need a new Integer? If that's the case `Integer.valueOf` is better

Reply:
Sure! I've repeated this mistake. Sorry :P

Reply:
Even simpler, i'm going to use a just an int.